sklearn/linear_model/_logistic.py

```diff
@@ -1036,7 +1042,7 @@ class LogisticRegression(BaseEstimator, LinearClassifierMixin,
 
     Parameters
     ----------
-    penalty : {'l1', 'l2', 'elasticnet', 'none'}, default='l2'
+    penalty :
```

Comment:
I do not think space before `:` is necessary then anymore, no?

Reply:
I think the space is a part of the [rst definition list spec](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#definition-lists)

>  Optional classifiers may follow the term on the same line, each after an inline " : " (space, colon, space). 

Now that I look at the spec, technically we do not need the `:` at all.

Reply:
No, that space is part of [numpydoc style guide](https://numpydoc.readthedocs.io/en/latest/format.html). [Google style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) is for example without it,  See [comparison](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/index.html#google-vs-numpy).

Because to my understanding, the internal Sphinx syntax expected is:

```
:param path: The path of the file to wrap
:type path: str
:param field_storage: The :class:`FileStorage` instance to wrap
:type field_storage: FileStorage
:param temporary: Whether or not to delete the file when the File
   instance is destructed
:type temporary: bool
:returns: A buffered writable file descriptor
:rtype: BufferedFileStorage
```

So I am guessing sklearn is using napoleon to convert numpy docstring to internal Sphinx one. So one should check how napoleon converts when there is no space before `:`?

(This is all me guessing. I do not know those details of sklearn.)

Reply:
> So I am guessing sklearn is using napoleon to convert numpy docstring to internal Sphinx one.

We are currently using numpydoc.

The space is picked up as a part of the parameter name:

### no space

<img width="143" alt="Screen Shot 2020-07-03 at 4 49 45 PM" src="https://user-images.githubusercontent.com/5402633/86496898-4b333500-bd4d-11ea-872e-185230ce614c.png">

### with space

<img width="142" alt="Screen Shot 2020-07-03 at 4 49 54 PM" src="https://user-images.githubusercontent.com/5402633/86496906-50907f80-bd4d-11ea-9a63-508978f77dcd.png">